src/include/sof/math/exp_fcn_hifi.h

```diff
@@ -0,0 +1,53 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2023 Intel Corporation. All rights reserved.
+ *
+ * Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+ *
+ */
+#ifndef __SOFM_EXP_HIFI_H__
+#define __SOFM_EXP_HIFI_H__
+
+#include <sof/platform.h>
+#include <stdint.h>
+#include <stddef.h>
+
+#define EXPONENTIAL_GENERIC 1
+#if defined(__XCC__)
+
+#include <xtensa/config/core-isa.h>
+#if XCHAL_HAVE_HIFI5 == 1
+#undef  EXPONENTIAL_GENERIC
+#undef EXPONENTIAL_HIFI3
```

Comment:
why not just have one define and check equality in your code? E.g.
EXPONENTIAL_HIFI == 5

Reply:
There's also no need to #undef something that has never been defined before, this is very verbose and hard to read.